hooks/_common.sh

```diff
@@ -163,21 +163,30 @@ function common::is_hook_run_on_whole_repo {
 # 2.1. If at least 1 check failed - change exit code to non-zero
 # 3. Complete hook execution and return exit code
 # Arguments:
-#   args (string with array) arguments that configure wrapped tool behavior
 #   hook_id (string) hook ID, see `- id` for details in .pre-commit-hooks.yaml file
+#   args_array_length (integer) Count of arguments in args array.
+#   args (array) arguments that configure wrapped tool behavior
 #   files (array) filenames to check
 #######################################################################
 function common::per_dir_hook {
-  local -r args="$1"
-  local -r hook_id="$2"
+  local -r hook_id="$1"
+  declare -i args_array_length=$2
   shift 2
+  declare -a args=()
+  # Expand args to a true array.
+  # Based on https://stackoverflow.com/a/10953834
+  while ((args_array_length-- > 0)); do
```

Comment:
Would a simpler code work here? This also makes no need for `args_array_length` var.
```suggestion
  while [[ $@ ]]; do
```
On the other hand the whole loop looks quite like simple `args=("$@")` 🤔

Reply:
Oh, I seem to think I understand why `args_array_length` is needed.
Though given `common::per_dir_hook "$HOOK_ID" "${ARGS[*]}" "${FILES[@]}"`, which is used across this PR, the 2nd ARG to this current function has ARGS array rather than number of its elements — is this expected? So it's either `common::per_dir_hook "$HOOK_ID" "${#ARGS[*]}" "${ARGS[*]}" "${FILES[@]}"`, or I am missing something essential 🤔 (probably it's right about time to leave this for tomorrow 😛)

Reply:
Okay, I couldn't leave it for tomorrow.
This `while` thing doesn't work as expected for me, and given `common::per_dir_hook "$HOOK_ID" "${ARGS[*]}" "${FILES[@]}"` this should be that all same thing from our discussion in Slack:
```
  while read -r -d '' ARG; do
    args+=("$ARG")
  done < <(echo "$1" | xargs printf '%s\0')
  shift
```

Reply:
No idea why, but the suggested solution does not work here 🤔 

[`7866b64` (#437)](https://github.com/antonbabenko/pre-commit-terraform/pull/437/commits/7866b64c3a1644bfd1049013329f381e379da190)

I try to replace `common::per_dir_hook "${ARGS[*]}"` with `common::per_dir_hook "${ARGS[@]}"` but got a weird error
```
dirname: unrecognized option '--version__REPLACED__SPACE__">__REPLACED__SPACE__1.1.8"'
Try 'dirname --help' for more information.
```

Also, no luck with changing `per_dir_hook_unique_part "${args[@]}"` to `per_dir_hook_unique_part "${args[*]}"`

Reply:
>Is the __REPLACED__SPACE__ to be replaced with actual space later down the code? 

Should be. Firstly, it replaces spaces with placeholders and then placeholders with spaces. Check file for `__REPLACED__SPACE__ `

Reply:
Nah, I mean is this expected that at this place the `__REPLACED__SPACE__` isn't replaced with spaces yet? This may influence how we should be parsing array elements since elements values are not space-separated string, but `__REPLACED__SPACE__`-separated string.